tests/slurm_plugin/test_clustermgtd.py

```diff
@@ -2112,6 +2170,794 @@ def test_update_status(self, compute_fleet_status_manager, put_item_response, ex
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", "launch_time"),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            False,
+            {"queue1-st-c5xlarge-1"},
+            "Node bootstrap error: Node is currently in replacement and no backing instance",
+            True,
+        ),
+        (
+            SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "some_state", "queue1"),
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", datetime(2020, 1, 1, 0, 0, 0)),
+            },
+            True,
+            {"queue1-st-c5xlarge-1"},
+            "Node bootstrap error: Replacement timeout expires for node in replacement",
+            True,
+        ),
+        (
+            SlurmNode("queue1-dy-c5xlarge-1", "ip-1", "hostname", "#MIXED", "queue1"),
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", "launch_time"),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            False,
+            {"some_node_in_replacement"},
+            "Node bootstrap error: Node is in power up state without valid backing instance",
+            True,
+        ),
+        (
+            SlurmNode("queue1-dy-c5xlarge-1", "ip-1", "hostname", "DOWN*+CLOUD+POWER", "queue1"),
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", "launch_time"),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            True,
+            {"some_node_in_replacement"},
+            "Node bootstrap error: Resume timeout expires",
+            True,
+        ),
+        (
+            SlurmNode("queue1-dy-c5xlarge-1", "ip-1", "hostname", "DOWN#+CLOUD", "queue1"),
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", "launch_time"),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            False,
+            {"some_node_in_replacement"},
+            None,
+            False,
+        ),
+        (
+            SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "DOWN*+CLOUD", "queue1"),
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", "launch_time"),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            False,
+            {"some_node_in_replacement"},
+            None,
+            False,
+        ),
+        (
+            SlurmNode("queue1-st-c5xlarge-1", "ip-1", "hostname", "DOWN*+CLOUD", "queue1"),
+            {
+                "ip-1": EC2Instance("id-1", "ip-1", "hostname", datetime(2020, 1, 1, 0, 0, 0)),
+                "ip-2": EC2Instance("id-2", "ip-2", "hostname", "launch_time"),
+            },
+            False,
+            {"some_node_in_replacement"},
+            None,
+            False,
+        ),
```

Comment:
These 2 test cases looks identical.
If we are testing run-instance failures, ip of the node should not be in list of instances